plugins/database/src/db_plugin.cpp

```diff
@@ -11901,6 +11901,47 @@ irods::error db_purge_server_load_digest_op(
 
 } // db_purge_server_load_digest_op
 
+irods::error db_set_delay_server_op(
+    irods::plugin_context& _ctx,
+    const char*            _hostname ) {
+    // =-=-=-=-=-=-=-
+    // check the context
+    irods::error ret = _ctx.valid();
+    if ( !ret.ok() ) {
+        return PASS( ret );
+    }
+
+    if ( logSQL != 0 ) {
+        rodsLog( LOG_SQL, "chlSetDelayServer" );
+    }
+
+    if ( _ctx.comm()->clientUser.authInfo.authFlag < LOCAL_PRIV_USER_AUTH ) {
+        return ERROR( CAT_INSUFFICIENT_PRIVILEGE_LEVEL, "insufficient privilege level" );
+    }
+
+    if ( !icss.status ) {
+        return ERROR( CATALOG_NOT_CONNECTED, "catalog not connected" );
+    }
+
+    int i = 0;
+    cllBindVars[i++] = _hostname;
+    cllBindVarCount = i;
+    if ( logSQL != 0 ) {
+        rodsLog( LOG_SQL, "chlSetDelayServer SQL 1" );
+    }
+
+    int status = cmlExecuteNoAnswerSql(
+                  "update R_GRID_CONFIGURATION set option_value = ? where namespace = 'delay_server' and option_name = 'successor'", &icss );
+    if ( status != 0 && status != CAT_SUCCESS_BUT_WITH_NO_INFO ) {
+        rodsLog( LOG_NOTICE,
+                 "chlSetDelayServer cmlExecuteNoAnswerSql failure %d" , status );
+        return ERROR( status, "Set Delay Server SQL update failure" );
+    }
+
```

Comment:
Do you need a "commit" call here?

Reply:
not sure.   autocommit?

Reply:
and, you are correct...

and with more success... i see this.

```
irods@9419ea2873d9:~$ psql ICAT -c 'select * from r_grid_configuration'
  namespace   |  option_name   | option_value 
--------------+----------------+--------------
 delay_server | leader         | 9419ea2873d9
 delay_server | successor      | 
 database     | schema_version | 9
(3 rows)

irods@9419ea2873d9:~$ iadmin set_delay_server bananas
irods@9419ea2873d9:~$ psql ICAT -c 'select * from r_grid_configuration'
  namespace   |  option_name   | option_value 
--------------+----------------+--------------
 delay_server | leader         | 9419ea2873d9
 database     | schema_version | 9
 delay_server | successor      | bananas
(3 rows)
```